src/Sylius/Behat/Context/Setup/ShippingContext.php

```diff
@@ -69,18 +80,30 @@ public function getShippingMethodByName($shippingMethodName)
 
     /**
      * @Given store ships everything for free
+     * @Given store ships everything for free within :zone zone
      */
-    public function storeShipsEverythingForFree()
+    public function storeShipsEverythingForFree(ZoneInterface $zone = null)
     {
-        $this->createShippingMethod('Free');
+        $this->createShippingMethod('Free', $zone);
     }
 
     /**
      * @Given /^store has "([^"]*)" shipping method with "(?:€|£|\$)([^"]*)" fee$/
+     * @Given /^store has "([^"]*)" shipping method with "(?:€|£|\$)([^"]*)" fee within ("([^"]*)" zone)$/
+     * @Given /^store has "([^"]*)" shipping method with "(?:€|£|\$)([^"]*)" fee for (the rest of the world)$/
+     */
+    public function storeHasShippingMethodWithFee($shippingMethodName, $fee, ZoneInterface $zone = null)
+    {
+        $this->createShippingMethod($shippingMethodName, $zone, 'en', ['amount' => $this->getFeeFromString($fee)]);
+    }
+
+    /**
+     * @Given /^(shipping method "[^"]+") belongs to ("[^"]+" tax category)$/
      */
-    public function storeHasShippingMethodWithFee($shippingMethodName, $fee)
+    public function shippingMethodBelongsToTaxCategory(ShippingMethodInterface $shippingMethod, TaxCategoryInterface $taxCategory)
     {
-        $this->createShippingMethod($shippingMethodName, 'en', ['amount' => $this->getFeeFromString($fee)]);
+        $shippingMethod->setTaxCategory($taxCategory);
+        $this->shippingMethodManager->flush($shippingMethod);
```

Comment:
We should not flush a specific entity as it usually leads to unexpected results because of missing cascading, etc.